### PR TITLE
Tehnalogeros/yea 258 part 1: your contributions, your archives, and archives lazy loading

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: frontend/components/Archives.tsx [hunk 1]
```diff
@@ -123,6 +100,7 @@ const Archives = () => {
           onClick={prevSlide}
           aria-label={"Previous"}
           isDisabled={startIndex <= 0}
+          backgroundColor={"transparent"}
         ></IconButton>
         <Flex w={slideAmount === 1 ? "200px" : "100%"}>
```
Comment (Ledespana) [inline frontend/components/Archives.tsx:105]: slideAmount now it should equal to 3 here, there may be more places

File: frontend/app/page.tsx [hunk 2]
```diff
@@ -101,45 +89,80 @@ export default function Home() {
...
-      {
-        label: "Archive Mints:",
-        value: "0".toLocaleString(),
-      },
-      { label: "Iters Till Next Archive:", value: "0".toLocaleString() },
-    ];
-  const gridTemplateColumns = { base: "repeat(1, 2fr)", md: "repeat(2, 1fr)" };
+  const fetchCollectionStats = async () => {
+    console.log("fetching collection stats");
+    const [totalSupply, iterations, contributors, lyxBurned] = await Promise.all([burntPixArchives.totalSupply(), burntPixArchives.getTotalIterations(), burntPixArchives.getTotalContributors(), burntPixArchives.getTotalFeesBurnt()]);
```
Comment (Ledespana) [inline frontend/app/page.tsx:94]: try catch error?

File: frontend/app/page.tsx [hunk 3]
```diff
@@ -101,45 +89,80 @@ export default function Home() {
...
-      const supplyCap = await burntPixArchives.tokenSupplyCap();
-      const lyxBurned = await burntPixArchives.getTotalFeesBurnt();
-
-      setIterations(iterations.toString());
-      setContributors(contributors.toString());
-      setArchiveMints(`${totalSupply.toString()} / ${supplyCap.toString()}`);
-      setLyxBurned(`${divideBigIntTokenBalance(lyxBurned, 18).toString()} LYX`);
+    const fetchImmutableStats = async () => {
+      console.log("fetching immutable data");
+      const [burntPicId, winnerIterations, supplyCap] = await Promise.all([burntPixArchives.burntPicId(), burntPixArchives.winnerIters(), burntPixArchives.tokenSupplyCap()]);
```
Comment (Ledespana) [inline frontend/app/page.tsx:106]: try catch error?

File: frontend/components/YourContributions.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        { label: "Iters Till Next Archive:", value: "--" },
+    ]);
+
+    const fetchUserStats = async (account: string) => {
+        if (!account) return;
+        console.log("fetching user stats", account)
+        const userIterations = await burntPixArchives.getContributions([account as AddressLike]);
+        const userArchives = await burntPixArchives.getArchives(account);
+        const userOwnedArchiveMints = await burntPixArchives.tokenIdsOf(account);
+        const userIterationsGoal = getNextIterationsGoal(userArchives.length + 1, Number(userIterations[0]));
```
Comment (Ledespana) [inline frontend/components/YourContributions.tsx:40]: try catch error?

File: frontend/app/page.tsx [hunk 5]
```diff
@@ -101,45 +89,80 @@ export default function Home() {
...
-    fetchStats();
+    fetchImmutableStats();
   }, []);
+
+  useEffect(() => {
+    supplyCap && fetchCollectionStats();
+  }, [supplyCap]);
+
+  const externalFetchArchives: IFetchArchives = useCallback(async (startFrom, amount, setArchives, setLoadedIndices, ownerProfiles, setOwnerProfiles) => {
+    const archiveCount = await burntPixArchives.archiveCount();
```
Comment (Ledespana) [inline frontend/app/page.tsx:120]: try catch error for promises. applies for rest of file/s

File: frontend/components/Archives.tsx [hunk 6]
```diff
@@ -29,81 +27,60 @@ interface IArchive {
...
-  const nextSlide = () => {
-    setStartIndex((prevIndex) => {
-      return Math.min(prevIndex + slideAmount, archives!.length - 1);
-    });
-  };
-
-  const prevSlide = () => {
-    setStartIndex((prevIndex) => Math.max(prevIndex - slideAmount, 0));
-  };
+  const slideAmount = useBreakpointValue({ base: 3, md: 5 }) || 5; // 1 image for base (mobile), 5 for md (tablet) and up
```
Comment (Ledespana) [inline frontend/components/Archives.tsx:59]: update comment

File: frontend/app/page.tsx [hunk 7]
```diff
@@ -101,45 +89,80 @@ export default function Home() {
...
+
+      newArchives.push({
+        id,
+        image: hexToText(archive.image),
+        ownerAddress,
+        ownerName: ownerProfile.name,
+        ownerAvatar,
+        isMinted,
+      });
+    }
```
Comment (democrazyfi) [inline frontend/app/page.tsx:157]: we should maybe this into the archives component as it's a lot of logic on the home page

was about to do the same for live stats but didn't want to conflict here

Comment (tehnalogos) [reply frontend/app/page.tsx:157]: yep. will work on refactor next stage as there's a lot here as is @democrazyfi

**Review by democrazyfi — APPROVED** [review]

**Pull request merged**